// http-auth/src/http-auth.js
"use strict";

// Utils.
const utils = require('http-auth/src/auth/utils');

// http integration.
require('http-auth/src/server/http');

// https integration.
require('http-auth/src/server/https');

// http-proxy integration.
if (utils.isAvailable('http-proxy')) {
    require('http-auth/src/server/proxy');
}

// Exports.
module.exports = {
    // Basic authentication.
    basic: (options, checker) => {
        return require('http-auth/src/auth/basic')(options, checker);
    },

    // Digest authentication.
    digest: (options, checker) => {
        return require('http-auth/src/auth/digest')(options, checker);
    },

    // Connect.
    connect: (auth) => {
        return require('http-auth/src/server/connect')(auth);
    },

    // Koa.
    koa: (auth) => {
        return require('http-auth/src/server/koa')(auth);
    },

    // Passport.
    passport: (auth) => {
        return require('http-auth/src/server/passport')(auth);
    },

    // Hapi.
    hapi: () => {
        return require('http-auth/src/server/hapi');
    }
};
